src/static/helpers/slasHelper.test.ts

```diff
@@ -0,0 +1,188 @@
+      clientId: '',
+      siteId: '',
+    },
+  },
+  authenticateCustomer: authenticateCustomerMock,
+  authorizeCustomer: authorizeCustomerMock,
+  getAccessToken: getAccessTokenMock,
+} as unknown as ShopperLogin<{
+  shortCode: string;
+  organizationId: string;
+  clientId: string;
+  siteId: string;
+}>;
+
+const mockParameters = {
+  codeVerifier: '',
+  shopperUserId: 'shopperUserID',
+  shopperPassword: 'shopperPassword',
+  redirectURI: '',
+};
+
+describe('Create code verifier', () => {
+  test('creates string greater between 43 and 128 char per standard', () => {
+    const verifier = slasHelper.createCodeVerifier();
+
+    expect(verifier.length).toBeGreaterThanOrEqual(43);
+    expect(verifier.length).toBeLessThanOrEqual(128);
+  });
+});
+
+describe('Generate code challenge', () => {
+  const verifier =
+    'XVv3DJSzPDdbcVsrcs-4KuUtMYhvd6fxS0Gtbu_gv-UaVKo80w8WKA1gitXhC-DMW0H_mOtUNJhecfTwb-n_dXQWz8Ay6iWZWoeSBPfwgzP_pblgQr4eqodqeYNxfdWv';
+  const expectedChallenge = 'AH8WaHxbEtoZuFw-rw2YS9SazhKJilGoESoSlICXsQw';
+
+  test('generates correct code challenge for verifier', async () => {
+    const challenge = await slasHelper.generateCodeChallenge(verifier);
+    expect(challenge).toBe(expectedChallenge);
+  });
+});
+
+describe('Get code and usid', () => {
+  const url =
+    'https://localhost:3000/callback?usid=048adcfb-aa93-4978-be9e-09cb569fdcb9&code=J2lHm0cgXmnXpwDhjhLoyLJBoUAlBfxDY-AhjqGMC-o';
+
+  const expectedRecord = {
+    code: 'J2lHm0cgXmnXpwDhjhLoyLJBoUAlBfxDY-AhjqGMC-o',
+    url,
+    usid: '048adcfb-aa93-4978-be9e-09cb569fdcb9',
+  };
+
+  test('extracts code and usid from url', () => {
+    const record = slasHelper.getCodeAndUsidFromUrl(url);
+    expect(record).toStrictEqual(expectedRecord);
+  });
+});
+
+describe('Authorize user', () => {
+  const expectedCode = 'J2lHm0cgXmnXpwDhjhLoyLJBoUAlBfxDY-AhjqGMC-o';
+
+  test('hits the authorize endpoint and recieves authorization code', async () => {
+    const authCode = (
+      await slasHelper.authorize(mockSlasClient, mockParameters)
+    ).code;
+    expect(authorizeCustomerMock).toHaveBeenCalled();
+    expect(authCode).toBe(expectedCode);
+  });
+});
+
+describe('Guest user flow', () => {
+  test('uses code challenge to generate auth code', async () => {
+    await slasHelper.loginGuestUser(mockSlasClient, mockParameters);
+
+    // Pulling out the arguments to test
+    const parametersArg = (
+      authorizeCustomerMock.mock.calls[0][0] as Parameters<
+        ShopperLogin<any>['authorizeCustomer'] // TODO: replace any with proper type
+      >[0]
+    )?.parameters;
+
+    // required query parameters
+    // expect(parametersArg?.hint).toBe('guest'); TODO: revise this line
+    expect(parametersArg?.response_type).toBe('code');
+    expect(parametersArg).toHaveProperty('client_id');
+    expect(parametersArg).toHaveProperty('code_challenge');
+    expect(parametersArg).toHaveProperty('redirect_uri');
+
+    expect(authorizeCustomerMock).toBeCalled();
```

Comment:
I'm not sure whether this test should focus on the method that is called or on the returned value. Depends on whether you consider the goal of the helper to be "makes and parses an HTTP request" or "calls an SDK method". If the former, the only the returned value should be checked, as the SDK method is a hidden implementation detail. If the latter, then using [`expect.objectContaining`](https://jestjs.io/docs/expect#expectobjectcontainingobject) provides a cleaner approach by allowing you to check against a subset of the called argument:
```suggestion
    expect(authorizeCustomerMock).toHaveBeenCalledWith({
      parameters: {
        response_type: 'code',
        // ...
      }
    });
```

Reply:
I prefer the test to focus on the latter "calls an SDK method"